### DISCUSSION TITLE
CUDA out of memory errors after several frames of batch input

### DISCUSSION THREAD

**Discussion by giacometti777** [thygate/stable-diffusion-webui-depthmap-script#23] [Ideas]
Thank you so much for developing this! A lot of my headspace is dedicated to extracting depth from SD outputs, video outputs etc but I am not technically capable so this is super liberating.

One issue with batch inputs - I am hitting CUDA memory errors after a certain number of frames because the GPU isn't clearing used VRAM. Is this something you could implement easily?

**Comment by thygate** [comment]
Hi, thanks. I'm pretty new to python, but it is my understanding the garbage collector frees memory as soon as the variable goes out of scope. That said, I'm aware that there must still be some memory leak somewhere as over long runs I also still get out of vram errors.
I'm still looking into it, so no fix yet ..

The big disadvantage with batch img2img is that the midas model needs to be reloaded every image.

If you cloned the midas repo into repositories/midas, you can run the `run.py` script in that directory directly from the command line, no need to use the webui, it will be much, much faster too, as the model does not have to be reloaded after each image .. 


1) Place one or more input images in the folder `input`.

2) Run the model:

    ```shell
    python run.py --model_type dpt_large
    python run.py --model_type dpt_hybrid 
    python run.py --model_type midas_v21_small
    python run.py --model_type midas_v21
    ```

3) The resulting inverse depth maps are written to the `output` folder.

**Reply by giacometti777** [reply]
Wow thank you so much for helping me with an actionable workflow. The world needs more people like you!

**Comment by thygate** [comment]
Should be solved now in v0.1.9
